sdk/state/open.go

```diff
@@ -2,19 +2,29 @@ package state
 
 import (
 	"fmt"
+	"strconv"
 
 	"github.com/stellar/experimental-payment-channels/sdk/txbuild"
 	"github.com/stellar/go/txnbuild"
 	"github.com/stellar/go/xdr"
 )
 
+// TODO - should store on channel like Update and Close proposals?
 type Open struct {
 	CloseSignatures       []xdr.DecoratedSignature
 	DeclarationSignatures []xdr.DecoratedSignature
 	FormationSignatures   []xdr.DecoratedSignature
+
+	Asset      Asset
+	AssetLimit string
+}
+
+type OpenParams struct {
```

Comment:
this is the main change of this PR. We're now passing the Asset to `ProposeOpen`, so that we can form the channel with the non-native asset

note: I added an input Params struct `OpenParams`, I'm not tied to the name bc sounds pretty similar to `Open`. But maybe different enough 🤔

Reply:
we could call it `AssetParams` or `AssetLimit` since it's encapsulating the Asset info like asset type and limit.
OpenParams is a little confusing, but adding a comment on the struct explaining it's intended use case allows us to be a little more flexible in naming it.

Reply:
left a comment. I think the params may be expanding to more than just asset later on so left as `OpenParams` for now
https://github.com/stellar/experimental-payment-channels/pull/91/commits/bafd563bb751648bfa3c240ee6415bc1790a820e